beacon-chain/rpc/prysm/v1alpha1/validator/proposer_builder.go

```diff
@@ -0,0 +1,93 @@
+package validator
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v3/beacon-chain/db/kv"
+	"github.com/prysmaticlabs/prysm/v3/config/params"
+	types "github.com/prysmaticlabs/prysm/v3/consensus-types/primitives"
+	"github.com/sirupsen/logrus"
+)
+
+// Returns true if builder (ie outsourcing block construction) can be used. Both conditions have to meet:
+// - Validator has registered to use builder (ie called registerBuilder API end point)
+// - Circuit breaker has not been activated (ie the liveness of the chain is healthy)
+func (vs *Server) canUseBuilder(ctx context.Context, slot types.Slot, idx types.ValidatorIndex) (bool, error) {
+	registered, err := vs.validatorRegistered(ctx, idx)
+	if err != nil {
+		return false, err
+	}
+	if !registered {
+		return false, nil
+	}
+	activated, err := vs.circuitBreakBuilder(slot)
```

Comment:
shouldn't we check this first ? if our circuit breaker had been triggered, we should not even check for validator registration.

Reply:
@nisdas that makes sense to me. Updated